api_db.php

```diff
@@ -158,7 +158,11 @@
 	{
 		$limit = "WHERE timestamp <= :until";
 	}
-	$stmt = $db->prepare('SELECT client,count(client) FROM queries '.$limit.' GROUP by client order by count(client) desc limit 20');
+    $dbquery= "Select CASE typeof(client) WHEN 'integer' THEN (";
+    $dbquery .= " SELECT CASE TRIM(name) WHEN '' THEN c.ip ELSE c.name END name FROM client_by_id c WHERE c.id = q.client)";
+    $dbquery .= " ELSE client END client, count(client) FROM query_storage q ".$limit." GROUP by client order by count(client) desc limit 20";
```

Comment:
Suggestion:
Use upper case for reserved words (for consistency).

```suggestion
    $dbquery = "SELECT CASE typeof(client) WHEN 'integer' THEN (";
    $dbquery .= " SELECT CASE TRIM(name) WHEN '' THEN c.ip ELSE c.name END name FROM client_by_id c WHERE c.id = q.client)";
    $dbquery .= " ELSE client END client, count(client) FROM query_storage q ".$limit." GROUP BY client ORDER BY count(client) DESC LIMIT 20";
```